### PR TITLE
Check trakt external id's, for shows that are already in the library.

### PR DESCRIPTION
**Pull request by p0psicles**
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

@duramato

### CODE REVIEW

File: medusa/show/recommendations/recommended.py [hunk 1]
```diff
@@ -130,9 +133,19 @@ def __init__(self, rec_show_prov, series_id, title, mapped_indexer, mapped_serie
...
+        indexers = {mapped_indexer: mapped_series_id}
+        indexers.update({VALID_INDEXERS[indexer_name]: indexers_series_id
+                         for indexer_name, indexers_series_id
+                         in self.ids.items() if indexer_name in VALID_INDEXERS})
+
+        self.show_in_list = False
+        for show in app.showList:
+            if show.indexer in indexers and show.series_id == indexers[show.indexer]:
+                self.show_in_list = True
+                self.mapped_indexer = show.indexer
```
Comment (p0psicles) [inline medusa/show/recommendations/recommended.py:145]: By default shows are added with tvdb. But if they're already in the db, we want to reference the indexer and series_id from the library.

File: medusa/show/recommendations/recommended.py [hunk 2]
```diff
@@ -42,6 +44,7 @@
 log.logger.addHandler(logging.NullHandler())
 
 session = MedusaSession()
+VALID_INDEXERS = {indexerConfig[indexer]['identifier']: indexer for indexer in indexerConfig}
```
Comment (medariox) [inline medusa/show/recommendations/recommended.py:47]: I feel like this should be defined in indexer config and imported. Why `VALID_INDEXERS`? What makes them valid?

Comment (p0psicles) [reply medusa/show/recommendations/recommended.py:47]: Indexers we use to add shows. Oppositie to imdb and trakt.

Comment (medariox) [reply medusa/show/recommendations/recommended.py:47]: I would keep this as `INDEXERS` and eventually the ones that only give info trakt, imdb as `INFO_INDEXERS`? Or even not call them indexers at all.

Comment (p0psicles) [reply medusa/show/recommendations/recommended.py:47]: `TV_SOURCE` ?

Comment (p0psicles) [reply medusa/show/recommendations/recommended.py:47]: I'll keep it as INDEXERS, else it's EXTERNALS

Comment (p0psicles) [reply medusa/show/recommendations/recommended.py]: I moved and renamed it. Put it in indexer.utils.

**Pull request merged**